resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -793,6 +793,25 @@ private[spark] class Client(
       // distributed file.
       amKeytabFileName.foreach { kt => props.setProperty(KEYTAB.key, kt) }
 
+      // Upload user provided ivysettings.xml file to the distributed cache
+      val ivySettings = sparkConf.getOption("spark.jars.ivySettings")
```

Comment:
it would be nice to move this into the configuration package.scala and use ConfigBuilder.  Even if we just reference it by the .key option in the SparkSubmitArguments file.

Reply:
Would it be okay to handle this as a followup immediately after this PR? There are about 7-8 other places where this string is hardcoded and we can also refactor them out into config package.scala

Reply:
that's fine.

Reply:
Filed https://issues.apache.org/jira/browse/SPARK-35074